### PR TITLE
Add anchors to show/hide sections in install page

### PR DESCRIPTION
**Pull request by abbycross**
EDIT: Adding custom anchor tags to the show/hide sections in the "Qiskit versioning" section.

Once this is merged, direct links to the show-hide sections will be:

https://docs.quantum.ibm.com/start/install#release-schedule
https://docs.quantum.ibm.com/start/install#upgrade-strategy
https://docs.quantum.ibm.com/start/install#pre-releases
https://docs.quantum.ibm.com/start/install#post-releases

---------------------------
(Original proposal below, for historical purposes; we are no longer separating out the versioning section to a new page.)
In discussions with @mtreinish, the proposal is to separate out the versioning section (currently at the end of the start/install page) into a new page, for the following reasons:

- it really isn't directly install-related
- it allows us to remove the show/hide features for that content, which makes it easier to link to and find

We may also want to remove the show/hide features for the OS support tiers, but leave them in the current install page, since they are install-related. 

Topics to discuss:

- [ ] What to put in Next steps section? Currently using the same as what is on the Install pg as a placeholder.
- [ ] Title at the moment is just Qiskit versioning (since that's the only thing in the new page); are we good with this?
- [ ] I put the new page after the "Advanced setup" section in the toc. Should it be elsewhere?

cc: @mtreinish @Eric-Arellano

### CODE REVIEW

**Comment by abbycross** [comment]
Ignore my first commit -- it's no longer the direction we're going in!!

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:128]: I think the only topics that apply as next steps would be Install and set up Qiskit, and Configure Qiskit locally.  And maybe Release notes (https://docs.quantum.ibm.com/api/qiskit/release-notes)

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:11]:
```suggestion
patch versions. For example, in version number `X.Y.Z`, `X` is the major version,
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:40]:
```suggestion
depend on Qiskit should not raise their minimum required Qiskit version to a new
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:49]:
```suggestion
(with an underscore `_` prefix). However, there can be explicit exceptions made for
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:50]:
```suggestion
specific documented APIs. In such cases, these APIs will be clearly documented
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:74]:
```suggestion
When a new major version is released, the recommended upgrade path
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:75]:
```suggestion
is to first upgrade to the most recent minor version on the previous major
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:77]:
```suggestion
be published. This final minor version release `X.Y+1.0.0` is equivalent to
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:78]:
```suggestion
`X.Y.0` but with warnings and deprecations for any API changes that are
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:87]:
```suggestion
Qiskit users should first upgrade to this final minor
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:88]:
```suggestion
version to see any deprecation warnings and adjust their Qiskit
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:89]:
```suggestion
usage before trying a potentially breaking release. The previous
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:91]:
```suggestion
to upgrade. A typical pattern to manage this is to pin the maximum version to
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:93]:
```suggestion
For example, specifying `qiskit<2` in a requirements file when the current
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:94]:
```suggestion
major Qiskit version is 1 ensures that you're using a version of Qiskit
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:95]:
```suggestion
that doesn't have breaking API changes.
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:97]:
```suggestion
Capping the version less than the next major version 
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:98]:
```suggestion
ensures that you see any deprecation warnings before a
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:99]:
```suggestion
major version release. 
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:100]:
```suggestion
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:101]:
```suggestion
Without the cap, `pip` installs 
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:102]:
```suggestion
the newest version available by default.
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:106]:
```suggestion
For each minor and major version release Qiskit publishes pre-releases that
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:119]:
```suggestion
If there are issues with a release's packaging, a post-release might be
```

File: docs/start/qiskit-versioning.mdx
Comment (beckykd) [inline docs/start/qiskit-versioning.mdx:121]:
```suggestion
integer indicates that it is the first post-release of the `X.Y.Z` release.
```

**Review by beckykd — APPROVED** [review]

File: docs/start/qiskit-versioning.mdx
Comment (abbycross) [inline docs/start/qiskit-versioning.mdx:133]:
```suggestion
  -  [Install and set up Qiskit.](install)
  -  [Configure Qiskit locally.](configure-qiskit-local)
  -  Visit the Qiskit [release notes.](/api/qiskit/release-notes) 
```

File: docs/start/install.mdx
Comment (Eric-Arellano) [inline docs/start/install.mdx:181]: Maybe link to the new page in Next Steps.

File: docs/start/qiskit-versioning.mdx
Comment (Eric-Arellano) [inline docs/start/qiskit-versioning.mdx:3]: Great description!

```suggestion
description: Description of how Qiskit version numbers work, and information about release schedules, upgrade strategy, pre-releases, and post-releases.
```

File: docs/start/qiskit-versioning.mdx
Comment (Eric-Arellano) [inline docs/start/qiskit-versioning.mdx:28]: Thanks for following the image file format <3

File: docs/start/_toc.json
Comment (Eric-Arellano) [inline docs/start/_toc.json:42]: I think this makes sense as top-level, rather than "Advanced setup". But I'm curious what others think.

**Review by Eric-Arellano — APPROVED** [review]
Looks great to me! Maybe wait for sign-off from Abby M

File: docs/start/_toc.json
Comment (javabster) [reply docs/start/_toc.json:42]: I'm not super keen on having it be a top level  page in the start section. We're really trying to minimise the amount of content there. what about moving this page back to the api reference? I remember we briefly discussed that last year, what do you think?

Comment (Eric-Arellano) [reply docs/start/_toc.json:42]: Where would that go in `/api`? A new section like `/api/open-source`?

**Comment by abbycross** [comment]
Update: @javabster suggested trying 

> `<summary id="anchor-name">Description words</summary>` 

within the `<details>` tags, and this works much better than `<span>`. In theory we can bring the show/hide sections back and keep everything on the same page -- however, we still do not have a way for the reader to copy an anchor's link (no little chain icon shows up). That alone may be reason enough to split out the content into a separate page; otherwise the reader would have to look at the code to discover the anchor tag. 

One option is to include a header within the show/hide content itself, which would show the chain icon, AND this means the header would show up on the right nav (if it's h2 or h3), but then we would have a duplicated line situation like this:
Screenshot 2024-01-10 at 2 51 29 PM



Another option is to create the header right within the summary tags, which would work great if the chain icon wasn't superimposed with the caret to click on!!
Screenshot 2024-01-10 at 2 47 01 PM

Thoughts or reasonable workarounds appreciated!... I have scoured the internet but perhaps am not using the right search terms...

**Comment by abbycross** [comment]
Had an offline conversation with @javabster and we determined that the best way to move forward is to return to the original formatting of this content and create custom anchor links for the show/hide sections. In the future we hope to have a better component for the show-hide sections so that we can more flexibly link to them.

**Comment by abbycross** [comment]
One last comment: while this functionality works, the links may appear only to jump to the Qiskit versioning section, because the show/hide section is so close to the end of the page. 🤷

**Pull request merged**
